compose/material3/adaptive/adaptive-layout/src/skikoMain/kotlin/androidx/compose/material3/adaptive/layout/Strings.skiko.kt

```diff
@@ -55,3 +44,85 @@ internal actual value class Strings(val value: Int) {
             get() = Strings(0)
     }
 }
+
+// TODO check if we should replace it by a more performant implementation
+//  (without creating intermediate strings)
+// TODO current implementation doesn't support sophisticated formatting like %.2f,
+//  but currently we use it only for integers and strings
+private fun String.format(vararg formatArgs: Any?): String {
+    var result = this
+    formatArgs.forEachIndexed { index, arg ->
+        result = result
+            .replace("%${index + 1}\$d", arg.toString())
+            .replace("%${index + 1}\$s", arg.toString())
+    }
+    return result
+}
+
+@Composable
+@ReadOnlyComposable
+internal actual fun getString(string: Strings): String {
+    val locale = Locale.current
+    val tag = localeTag(language = locale.language, region = locale.region)
+    val translation = translationByLocaleTag.getOrPut(tag) {
+        findTranslation(locale)
+    }
+    return translation[string] ?: error("Missing translation for $string")
+}
+
+@Composable
+@ReadOnlyComposable
+internal actual fun getString(string: Strings, vararg formatArgs: Any): String {
+    return getString(string).format(*formatArgs)
+}
+
+/**
+ * A single translation; should contain all the [Strings].
+ */
+internal typealias Translation = Map<Strings, String>
+
+/**
+ * Translations we've already loaded, mapped by the locale tag (see [localeTag]).
+ */
+private val translationByLocaleTag = mutableMapOf<String, Translation>()
+
+/**
+ * Returns the tag for the given locale.
+ *
+ * Note that this is our internal format; this isn't the same as [Locale.toLanguageTag].
+ */
+private fun localeTag(language: String, region: String) = when {
+    language == "" -> ""
+    region == "" -> language
+    else -> "${language}_$region"
+}
+
+/**
+ * Returns a sequence of locale tags to use as keys to look up the translation for the given locale.
+ *
+ * Note that we don't need to check children (e.g. use `fr_FR` if `fr` is missing) because the
+ * translations should never have a missing parent.
+ */
+private fun localeTagChain(locale: Locale) = sequence {
+    if (locale.region != "") {
+        yield(localeTag(language = locale.language, region = locale.region))
+    }
+    if (locale.language != "") {
+        yield(localeTag(language = locale.language, region = ""))
+    }
+    yield(localeTag("", ""))
+}
+
+/**
+ * Finds a [Translation] for the given locale.
+ */
+private fun findTranslation(locale: Locale): Map<Strings, String> {
+    // We don't need to merge translations because each one should contain all the strings.
+    return localeTagChain(locale).firstNotNullOf { translationFor(it) }
+}
+
+/**
+ * This object is only needed to provide a namespace for the [Translation] provider functions
+ * (e.g. [Translations.en]), to avoid polluting the global namespace.
+ */
+internal object Translations
```

Comment:
Please, run `./gradlew jbApiDump`.

Seems invisible `Translations$stableprop` is added